src/framework/main_control_loop.py

```diff
@@ -0,0 +1,59 @@
+from ..ControlTasks import ControlTaskBase, ClockManager, MissionManager, ReadCamera, PointTracker, ProcessFrame, DisplayFrame
+from ..sfr import StateFieldRegistry
+import time
+
+
+class MainControlLoop(ControlTaskBase):
```

Comment:
It would be helpful with classes like this to include which other classes or methods use it, and what methods are called by it- each individual class/method is really readable but it's hard to figure out how it all fits together